sqlglot/generator.py

```diff
@@ -4434,3 +4434,15 @@ def overlay_sql(self, expression: exp.Overlay):
     @unsupported_args("format")
     def todouble_sql(self, expression: exp.ToDouble) -> str:
         return self.sql(exp.cast(expression.this, exp.DataType.Type.DOUBLE))
+
+    def string_sql(self, expression: exp.String) -> str:
+        this = expression.this
+        zone = expression.args.get("zone")
+
+        if zone:
+            # BigQuery stores timestamps internally as UTC, so it's used as a default source timezone
```

Comment:
This comment may be somewhat confusing in the future, because this method is in the base generator. Can we do this in a source-dialect-agnostic way? Perhaps we can set a default zone at parse time?

Reply:
This is not doable afaict, because the `zone` arg here is the target timezone while `UTC` is the source. Also, BigQuery being the only variant with `zone` should hopefully make it less confusing, I can add more comments there if needed.

Reply:
Sounds good. Let's just point out that our current expectation is that this arg will only be set for BigQuery?